README.md

```diff
@@ -705,3 +705,187 @@ agentChatSession.cleanUpOnParticipantDisconnect();
+- "Enable real-time chat message streaming": [Documentation](https://docs.aws.amazon.com/connect/latest/adminguide/chat-message-streaming.html)
+
+### Walkthrough
+
+1) A new chat contact is initialized with a [`StartChatContact`](https://docs.aws.amazon.com/connect/latest/APIReference/API_StartChatContact.html) request
+
+```js
+const input = {
+  name: "Customer1",
+  apiGatewayEndpoint,
+  region,
+  contactFlowId,
+  instanceId,
+};
+
+// https://docs.aws.amazon.com/connect/latest/APIReference/API_StartChatContact.html
+const initiateChatRequest = {
+  ParticipantDetails: {
+    DisplayName: input.name
+  },
+  ContactFlowId: "string",
+  InstanceId: "string",
+  Attributes: JSON.stringify({
+    "customerName": customerName // pass this to the Contact flow
+  }),
+  SupportedMessagingContentTypes: ["text/plain", "text/markdown"]
+};
+
+// https://github.com/amazon-connect/amazon-connect-chat-ui-examples/tree/master/cloudformationTemplates/startChatContactAPI
+const initiateChat = () => {
+  return request(
+    input.apiGatewayEndpoint,
+    {
+      headers,
+      credentials: 'include',
+      method: 'post',
+      body: JSON.stringify(initiateChatRequest),
+    }
+  ).then((res) => res.json.data);
+}
+```
+
+2) The `ParticipantToken` is return from the `StartChatContact` response, which needs to be saved
+
+```http
+HTTP/1.1 200
+Content-type: application/json
+
+{
+  "ContactId": "string",
+  "ParticipantId": "string",
+  "ParticipantToken": "string"
+}
+```
+
+```js
+const startChatResult = await initiateChat({ /* ... */});
+
+const {
+  ContactId,
+  ParticipantId,
+  ParticipantToken // <--- SAVE THIS (eg. sessionStorage)
+} = startChatResult.chatDetails;
+```
+
+3) New chat session is established with `ChatSession.create()`, passing in `chatDetails` from `StartChatContact` response
+
+```js
+// Establish a new chat session
+connect.ChatSession.create({
+  chatDetails: {
+    ContactId,
+    ParticipantId,
+    ParticipantToken
+  },
+  type: "CUSTOMER",
+  // ...
+});
+```
+
+4) Browser is refreshed. Chat session is disconnected. Customer still wants to continue the conversation.
+
+5) A new chat contact is initialized on page load (`StartChatContact`)
+
+```js
+// Begin a new chat contact with a second StartChatContact request
+const startChatResultTwo = await initiateChat({ /* ... */ });
+
+const {
+  ContactId,
+  ParticipantId,
+} = startChatResultTwo.chatDetails;
+```
+
+6) The new chat session is established, passing the new `chatDetails` to ChatJS in `ChatSession.create()`. Pass in the previous `participantToken` from step 2.
+
+```js
+// Retreive the previos sessions `participantToken`
+const savedParticipantToken = "<PARTICIPANT_TOKEN>" // Could be saved in `sessionStorage`
+
+// Establish a new chat session
+connect.ChatSession.create({
+  chatDetails: {
+    chatDetailsTwo.ContactId,
```

Comment:
Did you manually test this? I think internally we pass all the three props from the response of the first StartChatContact call, not just the pToken. And the second StartChatContact call is not needed, internally we store the first response on the backend and return it in the second call without calling ACS again.
The CreateParticipantConnection only takes in pToken though, but I guess the contactId and participantId will be used for some purposes (e.g to determine message direction)

Reply:
Published new revision to address comments 👍 . Tested the steps and reworded instructions to be more helpful.

Second StartChatContact is def incorrec. Passing pToken to CreateParticipantConnection for sdk works, but ChatJS needs all three details.